src/snapred/backend/service/CalibrationService.py

```diff
@@ -191,27 +193,17 @@ def diffractionCalibration(self, request: DiffractionCalibrationRequest):
         focusScheme = focusName.split("_")[-1]
 
         # get the needed input data
-        groceryList = [
-            GroceryListItem(
-                workspaceType="nexus",
-                runNumber=request.runNumber,
-                loader="LoadEventNexus",
-                useLiteMode=request.useLiteMode,
-            ),
-            GroceryListItem(
-                workspaceType="grouping",
-                groupingScheme=focusScheme,
-                useLiteMode=request.useLiteMode,
-                instrumentPropertySource="InstrumentDonor",
-                instrumentSource="prev",
-            ),
-        ]
-        workspaceList = FetchGroceriesRecipe().executeRecipe(groceryList)["workspaces"]
-        groceries = {
-            "inputWorkspace": workspaceList[0],
-            "groupingWorkspace": workspaceList[1],
-        }
-        return DiffractionCalibrationRecipe().executeRecipe(ingredients, groceries)
+        self.groceryClerk.name("inputWorkspace").nexus(request.runNumber).useLiteMode(request.useLiteMode).add()
+        self.groceryClerk.name("groupingWorkspace").grouping(focusScheme).useLiteMode(
+            request.useLiteMode
+        ).fromPrev().add()
+        groceries = self.groceryService.fetchGroceryDict(self.groceryClerk.buildDict())
+        data = DiffractionCalibrationRecipe().executeRecipe(ingredients, groceries)
+
+        # save the calibration table
+        filename = f"{calpath}/SNAP_{request.runNumber}_difcal_{date.today().strftime('%Y%m%d')}.h5"
+        self.groceryService.writeDiffCalTable(filename, data["calibrationTable"])
```

Comment:
Saving of the diffcal table doesnt happen until after assessment.  Also paths generally wouldnt live here, that would be letting the implementation of the Data layer bleed into the service layer.  

I'm a bit uneasy being so specific as to mention nexus for similar reasons.

Reply:
I should probably change "nexus" to "neutrons" everywhere it occurs.  I was using it to indicate the workspace has neutron scattering data instead of a grouping definition (or other data).  I agree, confusing